File: src/main/java/org/cloudfoundry/credhub/view/RsaView.java
package org.cloudfoundry.credhub.view;

import org.cloudfoundry.credhub.credential.RsaCredentialValue;
import org.cloudfoundry.credhub.domain.RsaCredentialVersion;

@SuppressWarnings("unused")
public class RsaView extends CredentialView {

  RsaView() {  /* Jackson */ }

  RsaView(RsaCredentialVersion rsaCredential) {
    super(
      rsaCredential.getVersionCreatedAt(),
      rsaCredential.getUuid(),
      rsaCredential.getName(),
      rsaCredential.getCredentialType(),
      new RsaCredentialValue(rsaCredential.getPublicKey(), rsaCredential.getPrivateKey())
    );
  }
}

File: components/credentials/src/main/kotlin/org/cloudfoundry/credhub/views/RsaView.kt
package org.cloudfoundry.credhub.views

import org.cloudfoundry.credhub.credential.RsaCredentialValue
import org.cloudfoundry.credhub.domain.RsaCredentialVersion

class RsaView : CredentialView {
    internal constructor() : super() /* Jackson */ {}
    internal constructor(rsaCredential: RsaCredentialVersion) : super(
        rsaCredential.versionCreatedAt,
        rsaCredential.uuid,
        rsaCredential.name,
        rsaCredential.getCredentialType(),
        rsaCredential.metadata,
        RsaCredentialValue(rsaCredential.publicKey, rsaCredential.privateKey)
    ) {
    }
}
